README.md
# merlin_frpc

frp client for merlin


Changelog.txt
Koolshare Asuswrt-Merlin Frpc Changelog
===========================================
2.1.15 ( 2018-06-03 08:30 )
  - Update    程序：更新frp客户端到0.20.0版本
2.1.14 ( 2018-05-26 20:10 )
  - Update    程序：忘打包了，重新来一次
2.1.13 ( 2018-05-26 17:30 )
  - Update    程序：更新frp客户端到0.19.1版本，frp作者更新太频繁了，好累
2.1.12 ( 2018-05-13 13:00 )
  - Fix       脚本：修改privilege_token为token
2.1.11 ( 2018-05-11 15:00 )
  - Update    程序：更新frp客户端到0.18.0版本
2.1.10 ( 2018-04-03 13:00 )
  - Fix       Web：修复连接失败后的连接方式
  - Update    程序：更新frp客户端到0.16.1版本
2.1.9 ( 2018-01-30 14:30 )
  - Update    程序：更新frp客户端到0.16.0版本
2.1.8 ( 2018-01-24 12:00 )
  - Update    程序：更新frp客户端到0.15.1版本
2.1.7 ( 2018-01-19 11:00 )
  - Update    程序：更新frp客户端到0.15.0版本
  - Add       Web：增加当前配置文件内容查看
2.1.6 ( 2018-01-17 01:00 )
  - Fix       Web/脚本：修复启动脚本DDNS设置错误。
2.1.5 ( 2018-01-16 17:00 )
  - Add       Web/脚本：恢复旧配置方案，新旧方案并存。
  - Fix       Web/脚本：修复界面错误。
2.1.4 ( 2018-01-09 23:00 )
  - Update    程序：更新frp客户端到0.14.1版本
  - Update    Web/脚本：更新配置界面及对应设置脚本
2.1.3 ( 2017-9-21 20:55 )
  - Fix       脚本：修复开机启动的一些问题。
2.1.2 ( 2017-9-21 20:11 )
  - Fix       脚本：修复开机启动的一些问题。
2.1.1 ( 2017-9-18 11:08 )
  - Fix       web：修复web js错误。
2.1.0 ( 2017-09-16 22:30 )
  - Fix       脚本：修正脚本一些错误，最近脑袋不太够用。
2.0.9 ( 2017-09-13 14:43 )
  - Fix       脚本：修正脚本一些错误，原谅我的粗心吧。
2.0.8 ( 2017-09-13 11:10 )
  - Fix       脚本：修正脚本一些错误，全都是坑。
2.0.7 ( 2017-09-08 22:00 )
  - update    程序：更新frp客户端到0.13.0版本
  - Add       Web/脚本：增加 stcp(secret tcp)
2.0.6 ( 2017-07-16 19:56 )
  - Add       Web/脚本：增加进程运行状态显示
2.0.5 ( 2017-07-02 09:56 )
  - Fix       Web/脚本：修复Frp和路由自带的DDNS冲突问题
  - Update    Web/脚本：允许TCP和UDP端口域名为空
2.0.4 ( 2017-06-27 00:25 )
  - update    Web：更新客户端连接失败操作
2.0.3 ( 2017-06-27 00:00 )
  - update    程序：更新frp客户端到0.12.0版本
2.0.2 ( 2017-04-09 11:32 )
  - Fix       Web：修复添加SSH穿透的本地地址错误。
2.0.1 ( 2017-04-09 01:05 )
  - Fix       脚本：修复错误。
2.0.0 ( 2017-04-09 00:12 )
  - Add       Web：预置的路由器穿透配置，简化路由器穿透配置内容，路由器http穿透只需要填写域名就行。
1.9.8 ( 2017-02-03 11:05 )
  - Fix       好吧，忘了把更新的页面上传了，重新来过
1.9.7 ( 2017-02-03 10:44 )
  - update    Web：增加UDP协议穿透
  - update    Web：紧随S大脚步移除显示密码勾选框，现在鼠点击即可显示
  - Fix       Web：修复提交按钮后默认TCP协议远程端口不可填写问题
1.9.6 ( 2017-02-02 16:23 )
  - update    程序：更新frp客户端到0.9.3版本
1.9.5 ( 2016-12-19 17:03 )
  - update    界面：哇哈哈哈，修复更新说明连接
1.9.4 ( 2016-12-19 17:03 )
  - update    好吧，我就是刷个版本号
1.9.3 ( 2016-12-18 13:17 )
  - update    脚本：修复不能自动启动问题
1.9.2 ( 忘了 )
  - update    脚本：修改启动脚本；
1.9.1 ( 2016-10-12 19:31 )
  - update    界面：添加https支持，现在https模式下也能获取更新版本信息了；
1.9.0 ( 2016-10-11 09:02 )
  - update    脚本：为frpc添加GOGC=40，以限制内存占用
1.8.1 ( 2016-10-05 21:50 )
  - Add       增加密码显示功能
1.8.0 ( 2016-10-05 21:50 )
  - Add       增加网络地图中的DDNS显示设置
  - Fix       修复一些脚本BUG
1.7.0 ( 2016-10-04 22:22 )
  - Add       增加表单验证功能
1.6.0 ( 2016-10-02 13:12 )
  - Add       设置界面新增定时注册服务功能
  - Fix       修复一些脚本BUG
1.5.0 ( 2016-10-01 23:20 )
  - Fix       修复开机启动
1.4.0 ( 2016-10-01 22:05 )
  - Add       增加定时启动功能
1.3.0 ( 2016-10-01 19:41 )
  - Fix       修改远程端口设置
  - Fix       修复启动脚本错误
1.2.0 ( 2016-10-01 18:27 )
  - Add       添加编辑现有穿透项目
  - Add       添加服务器相关设置
1.1.0 ( 2016-10-01 10:22 )
  - Fix       呃，忘了修复的内容了
1.0.0 ( 2016-09-30 23:12 )
  - Add       添加Frp穿透神器


frpc/res/frpc-menu.js
function openssHint(itemNum){
    statusmenu = "";
    width="350px";

    if(itemNum == 0){
        statusmenu ="如果发现开关不能开启，那么请检查<a href='Advanced_System_Content.asp'><u><font color='#00F'>系统管理 -- 系统设置</font></u></a>页面内Enable JFFS custom scripts and configs是否开启。";
        _caption = "服务器说明";
    }
    else if(itemNum == 1){
        statusmenu ="此处填入你的frp服务器的地址。</br>建议优先填入<font color='#F46'>IP地址</font>。填入域名，特别是一些服务商给的复杂域名，有时遇到无法解析会导致无法连接!";
        _caption = "服务器";
    }
    else if(itemNum == 2){
        statusmenu ="此处填入你的frp服务器的端口，对应服务器配置文件中的节[common]下的bind_port字段";
        _caption = "服务器端口";
    }
    else if(itemNum == 3){
        statusmenu ="此处填入你的frp服务器的特权授权码。对应服务器配置文件中的节[common]下的privilege_token字段。</br><font color='#F46'>注意：</font>使用带有特殊字符的密码，可能会导致frpc链接不上服务器。";
        _caption = "Privilege Token";
    }
    else if(itemNum == 4){
        statusmenu ="此处填入你的frp服务器HTTP穿透服务的端口，对应服务器配置文件中的节[common]下的vhost_http_port字段";
        _caption = "HTTP穿透服务端口";
    }
    else if(itemNum == 5){
        statusmenu ="此处填入你的frp服务器HTTPS穿透服务的端口，对应服务器配置文件中的节[common]下的vhost_https_port字段";
        _caption = "HTTPS穿透服务端口";
    }
    else if(itemNum == 6){
        statusmenu ="此处是否开启frpc客户端日志。</br><font color='#F46'>注意：</font>默认不开启，开启后日志路径为/tmp/frpc.log";
        _caption = "日志记录";
    }
    else if(itemNum == 7){
        statusmenu ="此处选择日志记录等级。</br>可选内容：info、warn、error、debug。";
        _caption = "日志等级";
    }
    else if(itemNum == 8){
        statusmenu ="此处选择要保留的日志天数。";
        _caption = "日志记录天数";
    }
    else if(itemNum == 9){
        statusmenu ="要穿透的协议类型，目前有http和tcp两种方式。";
        _caption = "协议类型";
    }
    else if(itemNum == 10){
        statusmenu ="此处输入穿透内容的命名（描述），如：ac68u-web或ac68u-webshell，对应客户端配置文件中的节名称。</br><font color='#F46'>注意：</font>frp服务器上的所有命名不能重复！";
        _caption = "代理名称";
    }
    else if(itemNum == 11){
        statusmenu ="此处输入穿透内容的域名，如：ac68u.frp.com，对应客户端配置文件中节下的custom_domains字段。</br><font color='#F46'>注意：</font>frp上运行的域名不能重复！";
        _caption = "域名配置";
    }
    else if(itemNum == 12){
        statusmenu ="此处输入要穿透的内部主机IP地址，如：192.168.1.1";
        _caption = "内网主机地址";
    }
    else if(itemNum == 13){
        statusmenu ="此处输入要穿透的内部主机的端口，如：80或22";
        _caption = "内网主机端口";
    }
    else if(itemNum == 14){
        statusmenu ="此处输入服务器端端口用来映射内部主机端口，如：80或8080</br><font color='#F46'>注意：</font>";
        statusmenu +="</br><b><font color='#669900'>http协议：</font></b>选择http协议时，远程主机端口对应服务器配置文件中的节[common]下的vhost_http_port字段值。";
        statusmenu +="</br><b><font color='#669900'>https协议：</font></b>选择https协议时，远程主机端口对应服务器配置文件中的节[common]下的vhost_https_port字段值。https协议只能对应穿透内网https协议。";
        statusmenu +="</br><b><font color='#669900'>tcp协议：</font></b>选择tcp协议时，远程主机端口应在服务器配置文件中的节[common]下的privilege_allow_ports字段值范围内。";
        _caption = "远程主机端口";
    }
    else if(itemNum == 15){
        statusmenu ="如果公司内网防火墙对外网访问进行了流量识别与屏蔽，例如禁止了 ssh 协议等，通过设置加密，将 frpc 与 frps 之间的通信内容加密传输，将会有效防止流量被拦截。";
        _caption = "加密";
    }
    else if(itemNum == 16){
        statusmenu ="如果传输的报文长度较长，通过设置对传输内容进行压缩，可以有效减小 frpc 与 frps 之间的网络流量，加快流量转发速度，但是会额外消耗一些 cpu 资源。";
        _caption = "压缩";
    }
    else if(itemNum == 17){
        statusmenu ="定时到Frp服务器上重新注册服务，以便Frp提供持续的服务。</br><font color='#F46'>注意：</font>填写内容为0时关闭该功能！";
        _caption = "定时注册服务";
    }
    else if(itemNum == 18){
        statusmenu ="如果穿透服务配置中内网主机地址是路由器管理地址，并且内网主机端口为80时，在网络地图DDNS处显示相应的域名配置。</br><font color='#F46'>注意：</font>此功能与路由系统自带的DDNS功能冲突，frp的DDNS显示设置会覆盖系统自带的DDNS设置！";
        _caption = "DDNS显示设置";
    }
    else if(itemNum == 19){
        statusmenu ="穿透服务的用户名称，如AC68U 配置结果为 {用户名称}.{代理名称}";
        _caption = "Frpc用户名称";
    }
    else if(itemNum == 20){
        statusmenu ="从 v0.12.0 版本开始，底层通信协议支持选择 kcp 协议，在弱网环境下传输效率提升明显，但是会有一些额外的流量消耗";
        _caption = "底层通信协议";
    }
    else if(itemNum == 21){
        statusmenu ="从 v0.10.0 版本开始，客户端和服务器端之间的连接支持多路复用，不再需要为每一个用户请求创建一个连接，使连接建立的延迟降低，并且避免了大量文件描述符的占用，使 frp 可以承载更高的并发数。</br>该功能默认启用，如需关闭，可以在 frps.ini 和 frpc.ini 中配置，该配置项在服务端和客户端必须一致.";
        _caption = "TCP 多路复用";
    }
    else if(itemNum == 22){
        statusmenu ="当客户端连接服务器失败后的动作：</br>失败后重复连接</br>失败后退出客户端";
        _caption = "连接设置";
    }
        //return overlib(statusmenu, OFFSETX, -160, LEFT, DELAY, 200);
        //return overlib(statusmenu, OFFSETX, -160, LEFT, STICKY, WIDTH, 'width', CAPTION, " ", FGCOLOR, "#4D595D", CAPCOLOR, "#000000", CLOSECOLOR, "#000000", MOUSEOFF, "1",TEXTCOLOR, "#FFF", CLOSETITLE, '');
        return overlib(statusmenu, OFFSETX, -160, LEFT, STICKY, WIDTH, 'width', CAPTION, _caption, CLOSETITLE, '');

    var tag_name= document.getElementsByTagName('a');
    for (var i=0;i<tag_name.length;i++)
        tag_name[i].onmouseout=nd;

    if(helpcontent == [] || helpcontent == "" || hint_array_id > helpcontent.length)
        return overlib('<#defaultHint#>', HAUTO, VAUTO);
    else if(hint_array_id == 0 && hint_show_id > 21 && hint_show_id < 24)
        return overlib(helpcontent[hint_array_id][hint_show_id], FIXX, 270, FIXY, 30);
    else{
        if(hint_show_id > helpcontent[hint_array_id].length)
            return overlib('<#defaultHint#>', HAUTO, VAUTO);
        else
            return overlib(helpcontent[hint_array_id][hint_show_id], HAUTO, VAUTO);
    }
}
